plugins/modules/identity/keycloak/keycloak_realm.py

```diff
@@ -0,0 +1,815 @@
+    realm_params = [x for x in module.params
+                    if x not in list(keycloak_argument_spec().keys()) + ['state'] and
+                    module.params.get(x) is not None]
+    keycloak_argument_spec().keys()
+    # See whether the realm already exists in Keycloak
+    before_realm = kc.get_realm_by_id(realm=realm)
+
+    if before_realm is None:
+        before_realm = dict()
+
+    # Build a proposed changeset from parameters given to this module
+    changeset = dict()
+
+    for realm_param in realm_params:
+        new_param_value = module.params.get(realm_param)
+
+        # some lists in the Keycloak API are sorted, some are not.
+        if isinstance(new_param_value, list):
+            if realm_param in ['attributes']:
+                try:
+                    new_param_value = sorted(new_param_value)
+                except TypeError:
+                    pass
+        # Unfortunately, the ansible argument spec checker introduces variables with null values when
+        # they are not specified
+        if realm_param == 'protocol_mappers':
+            new_param_value = [dict((k, v) for k, v in x.items() if x[k] is not None) for x in new_param_value]
+
+        changeset[camel(realm_param)] = new_param_value
+
+    # Whether creating or updating a realm, take the before-state and merge the changeset into it
+    updated_realm = before_realm.copy()
+    updated_realm.update(changeset)
+
+    result['proposed'] = sanitize_cr(changeset)
+    result['existing'] = sanitize_cr(before_realm)
+
+    # If the realm does not exist yet, before_realm is still empty
+    if before_realm == dict():
+        if state == 'absent':
+            # do nothing and exit
+            if module._diff:
+                result['diff'] = dict(before='', after='')
+            result['msg'] = 'Realm does not exist, doing nothing.'
+            module.exit_json(**result)
+
+        # create new realm
+        result['changed'] = True
+        if 'id' not in updated_realm:
+            module.fail_json(msg='id needs to be specified when creating a new realm')
+
+        if module._diff:
+            result['diff'] = dict(before='', after=sanitize_cr(updated_realm))
+
+        if module.check_mode:
+            module.exit_json(**result)
+
+        kc.create_realm(updated_realm)
+        after_realm = kc.get_realm_by_id(updated_realm['id'])
+
+        result['end_state'] = sanitize_cr(after_realm)
+
+        result['msg'] = 'Realm %s has been created.' % updated_realm['id']
+        module.exit_json(**result)
+    else:
+        if state == 'present':
+            # update existing realm
+            result['changed'] = True
+            if module.check_mode:
+                # We can only compare the current realm with the proposed updates we have
+                if module._diff:
+                    result['diff'] = dict(before=sanitize_cr(before_realm),
```

Comment:
You have already sanitized `before_realm` in line 741:
```suggestion
                    result['diff'] = dict(before=sanitize_cr(before_realm),
```

Reply:
There are a couple of places below that do the same thing.